Assignment1Q1.java

```diff
@@ -0,0 +1,110 @@
+        if (!validate(str1, str2)) {
+            return false;
+        }
+        if (str1.isEmpty() && str2.isEmpty()) { // simple case
+            return true;
+        }
+        str1 = str1.toLowerCase();
+        str2 = str2.toLowerCase();
+
+        // we can use array instead of hash map
+        // because we definitely know amount of characters
+        int[] counter = new int[CHARS_AMOUNT];
+        int startCode = Character.codePointAt("a", 0);
+        for (int i = 0; i < str1.length(); i++) {
+            int idx = str1.codePointAt(i) - startCode;
+            if (idx < 0 || idx >= CHARS_AMOUNT) {
+                throw new IllegalArgumentException("Unsupported character.");
+            }
+            counter[idx]++;
+        }
+        // no need to create another array for checking
+        for (int i = 0; i < str2.length(); i++) {
+            int idx = str2.codePointAt(i) - startCode;
+            if (idx < 0 || idx >= CHARS_AMOUNT) {
+                throw new IllegalArgumentException("Unsupported character.");
+            }
+            int checkValue = --counter[idx];
+            if (checkValue < 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    public static void main(String[] args) {
+        System.out.println(isEnglishAlphabetPermutation(null, null));            // false
+        System.out.println(isEnglishAlphabetPermutation(null, ""));              // false
+        System.out.println(isEnglishAlphabetPermutation("", ""));                // true
+        System.out.println(isEnglishAlphabetPermutation("a", "a"));              // true
+        System.out.println(isEnglishAlphabetPermutation("a", "ab"));             // false
+        System.out.println(isEnglishAlphabetPermutation("a", "A"));              // true
+        System.out.println(isEnglishAlphabetPermutation("abc", "ACB"));          // true
+        System.out.println(isEnglishAlphabetPermutation("abc", "ABB"));          // false
+        System.out.println(isEnglishAlphabetPermutation("abc", "abd"));          // false
+        System.out.println();
+
+        System.out.println(isPermutation(null, null));            // false
+        System.out.println(isPermutation(null, ""));              // false
+        System.out.println(isPermutation("", ""));                // true
+        System.out.println(isPermutation("a", "a"));              // true
+        System.out.println(isPermutation("a", "ab"));             // false
+        System.out.println(isPermutation("a", "A"));              // true
+        System.out.println(isPermutation("abc", "ACB"));          // true
+        System.out.println(isPermutation("abc", "ABB"));          // false
+        System.out.println(isPermutation("abc", "abd"));          // false
+        System.out.println(isPermutation("ab.", "ab/"));          // false
+        System.out.println(isPermutation("...55,,", ".5.5,.,"));  // true

```

Comment:
Great set of tests. Others you could have considered:
- A string containing 'Z' / 'z', as it is the last value of the array (boundary condition)
- Strings containing '@', '[', '`' or '(' as they are the boundary conditions for 'a', 'A', 'z' and 'Z' in ASCII
- compare an empty string to a non-empty string